### DISCUSSION TITLE
Magisk Alpha instead of Magisk

### DISCUSSION THREAD

**Discussion by MohamedAliClai** [LSPosed/MagiskOnWSALocal#216] [Q&A] [answered]
Hi
is there any way to use magisk alpha instead of official magisk?
that's because in Magisk alpha still option to use Magiskhide or Zygisk Denylist.
on the android phone I just flash it via TWRP, but here in WSA I don't know how to do this.

Thanks :)

**Comment by Howard20181** [comment, marked as answer]
Rename it to magisk-debug.zip and put it in $DOWNLOAD_DIR
run `build.sh --magisk-custom`

**Reply by MohamedAliClai** [reply]
what is the path of $DOWNLOAD_DIR

**Reply by Howard20181** [reply]
You'll know once you build with the default configuration

**Reply by MohamedAliClai** [reply]
do you mean that ?
\\wsl.localhost\Ubuntu\mnt\wslg\distro\home\cena_\MagiskOnWSALocal\download
if yes, I can't add any file to it, gives me remove the write protection.

**Reply by Howard20181** [reply]
`\\wsl.localhost\Ubuntu\home\cena_\MagiskOnWSALocal\download`

**Reply by MohamedAliClai** [reply]
I Renamed it to magisk-debug.zip and put it in `\\wsl.localhost\Ubuntu\home\cena_\MagiskOnWSALocal\download`
but when I run `build.sh --magisk-custom` it says `Custom Magisk not found, please rename it to magisk-debug.zip and put it in ../download`

![image](https://user-images.githubusercontent.com/101479148/197484060-a14c925d-788e-4d90-b21f-5f7fb98c9794.png)

**Reply by Howard20181** [reply]
Obviously you are not in the same working directory
`/mnt/c/Users/cena_/MagiskOnWSALocal/download`

**Reply by Howard20181** [reply]
`C:\Users\cena_\MagiskOnWSALocal\download`

**Reply by MohamedAliClai** [reply]
thanks for your support
it works for me 
but it may be something wrong in Magisk alpha making it doesn't work in WSA.

**Reply by Howard20181** [reply]
Find a bug in the original version, which will cause alpha to crash. Wait for the bug to fix.
https://github.com/topjohnwu/Magisk/issues/6353

**Reply by Howard20181** [reply]
Fixed in 44643ad7-alpha 25205
